src/Microsoft.Extensions.Configuration.FileExtensions/FileConfigurationProvider.cs

```diff
@@ -42,7 +43,8 @@ private void Load(bool reload)
                 }
                 else
                 {
-                    throw new FileNotFoundException($"The configuration file '{Source.Path}' was not found and is not optional.");
+                    var path = file?.PhysicalPath ?? Source.Path;
+                    throw new FileNotFoundException($"The configuration file '{path}' was not found and is not optional.");
```

Comment:
Should we always show the "virtual path", and then if the physical path is available, *also* show that?